### PR TITLE
fix nil hearing location in emails

### LINKED ISSUE #14188: VirtualHearingMailer fails to send email when hearing location is nil
## Description
Some emails for virtual hearings attached to legacy hearings are failing to send because we assume that a hearing's `hearing_location` will not be nil in several of our email templates. The error is caught in **[VirtualHearing::SendEmail](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/jobs/virtual_hearings/send_email.rb#L100)**.

## Acceptance criteria
- [ ] tests

## Technical notes
**[Slack thread](https://dsva.slack.com/archives/C01155MJGF2/p1588626846011000)** for context.

### PR DESCRIPTION
**Pull request by rubaiyat22**
Resolves #14188

### Description
- write a new method `hearing_location_or_regional_office` which defaults to `regional_office` for virtual hearing emails if `hearing_location` is `nil`
- write a method `full_address` in `HearingLocation` and `RegionalOffice` models which returns the full location 
- use `full_address` in emails
- write tests

### Acceptance Criteria
- [x] tests

### CODE REVIEW

File: app/models/hearings/forms/base_hearing_update_form.rb [hunk 1]
```diff
@@ -85,8 +85,11 @@ def start_async_job?
   end
 
   def start_async_job
-    start_cancel_job if start_async_job? && virtual_hearing_cancelled?
-    start_activate_job if start_async_job?
+    if start_async_job? && virtual_hearing_cancelled?
```
Comment (rubaiyat22) [inline app/models/hearings/forms/base_hearing_update_form.rb:88]: unrelated change: I made this change so only or the other job runs, not sure if it makes a difference

File: spec/models/concerns/has_virtual_hearing_examples.rb [hunk 2]
```diff
@@ -14,4 +14,22 @@
...
+    context "hearing location is nil" do
+      it "returns regional office" do
+        instance_of_class.update!(hearing_location: nil)
+        expect(subject).to eq(instance_of_class.reload.regional_office)
+      end
+    end
+
+    context "hearing location is not nil" do
+      it "returns hearing location" do
+        # binding.pry
```
Comment (ferristseng) [inline spec/models/concerns/has_virtual_hearing_examples.rb:30]: Commented out `binding.pry`

File: spec/mailers/virtual_hearing_mailer_spec.rb [hunk 3]
```diff
@@ -239,6 +239,41 @@
...
+    describe "hearing_location is not nil" do
+      it "shows correct hearing location" do
+        expect(subject.html_part.body).to include(hearing.location.full_address)
+        expect(subject.html_part.body).to include(hearing.hearing_location.name)
+      end
+    end
+
+    describe "hearing_location is nil" do
+      it "shows correct hearing location" do
+        hearing.update!(hearing_location: nil)
```
Comment (ferristseng) [inline spec/mailers/virtual_hearing_mailer_spec.rb:252]: Is this the best way to set the `hearing_location` to `nil`? I think it should be possible to do using the factory

File: app/models/hearing_location.rb [hunk 4]
```diff
@@ -15,32 +15,47 @@ def name
...
 
+  def key
+    vba_372? ? "VACO" : facility_id
+  end
+
   def vba_372?
     facility_id == "vba_372"
   end
+
+  def facility_address(key)
```
Comment (ferristseng) [inline app/models/hearing_location.rb:58]: Small nit to rename this param, since it's the same as the `key` method, and ends up reading a bit confusing to me.

File: app/models/concerns/has_virtual_hearing.rb [hunk 5]
```diff
@@ -23,4 +23,8 @@ def hearing_request_type
 
     readable_request_type
   end
+
+  def hearing_location_or_regional_office
+    location.nil? ? regional_office : location
+  end
```
Comment (ferristseng) [inline app/models/concerns/has_virtual_hearing.rb:29]: Could we move this to a `HearingLocationConcern` or something? This doesn't necessarily seem to me to be tied to just the virtual hearing. One other possible place we could use this is in the hearing serializer (https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/serializers/api/v2/hearing_serializer.rb)

File: spec/models/concerns/has_virtual_hearing_examples.rb [hunk 6]
```diff
@@ -14,4 +14,22 @@
       it { expect(subject).to be false }
     end
   end
+
+  context ".hearing_location_or_regional_office" do
+    subject { instance_of_class.reload.hearing_location_or_regional_office }
+
+    context "hearing location is nil" do
+      it "returns regional office" do
+        instance_of_class.update!(hearing_location: nil)
```
Comment (ferristseng) [inline spec/models/concerns/has_virtual_hearing_examples.rb:23]: Same here with settings `hearing_location` to `nil`

File: spec/mailers/virtual_hearing_mailer_spec.rb (hunk 3 above)
Comment (rubaiyat22) [reply spec/mailers/virtual_hearing_mailer_spec.rb:252]: Here I wanted to use the shared contexts which already have hearing_locations
